tgui/packages/tgui/interfaces/StripMenu.tsx

```diff
@@ -0,0 +1,394 @@
+  },
+
+  wear_r_ear: {
+    displayName: 'right earwear',
+    gridSpot: getGridSpotKey([0, 3]),
+    image: 'inventory-ears.png',
+  },
+
+  wear_l_ear: {
+    displayName: 'left earwear',
+    gridSpot: getGridSpotKey([1, 3]),
+    image: 'inventory-ears.png',
+  },
+
+  handcuffs: {
+    displayName: 'handcuffs',
+    gridSpot: getGridSpotKey([1, 4]),
+    hideEmpty: true,
+  },
+
+  legcuffs: {
+    displayName: 'legcuffs',
+    gridSpot: getGridSpotKey([1, 5]),
+    hideEmpty: true,
+  },
+
+  w_uniform: {
+    displayName: 'uniform',
+    gridSpot: getGridSpotKey([2, 1]),
+    image: 'inventory-uniform.png',
+  },
+
+  wear_suit: {
+    displayName: 'suit',
+    gridSpot: getGridSpotKey([2, 2]),
+    image: 'inventory-suit.png',
+  },
+
+  gloves: {
+    displayName: 'gloves',
+    gridSpot: getGridSpotKey([2, 3]),
+    image: 'inventory-gloves.png',
+  },
+
+  r_hand: {
+    displayName: 'right hand',
+    gridSpot: getGridSpotKey([2, 4]),
+    image: 'inventory-hand_r.png',
+    additionalComponent: <CornerText align="left">R</CornerText>,
+  },
+
+  l_hand: {
+    displayName: 'left hand',
+    gridSpot: getGridSpotKey([2, 5]),
+    image: 'inventory-hand_l.png',
+    additionalComponent: <CornerText align="right">L</CornerText>,
+  },
+
+  shoes: {
+    displayName: 'shoes',
+    gridSpot: getGridSpotKey([3, 2]),
+    image: 'inventory-shoes.png',
+  },
+
+  j_store: {
+    displayName: 'suit storage item',
+    gridSpot: getGridSpotKey([4, 0]),
+    image: 'inventory-suit_storage.png',
+  },
+
+  id: {
+    displayName: 'ID',
+    gridSpot: getGridSpotKey([4, 1]),
+    image: 'inventory-id.png',
+  },
+
+  belt: {
+    displayName: 'belt',
+    gridSpot: getGridSpotKey([4, 2]),
+    image: 'inventory-belt.png',
+  },
+
+  back: {
+    displayName: 'backpack',
+    gridSpot: getGridSpotKey([4, 3]),
+    image: 'inventory-back.png',
+  },
+
+  l_store: {
+    displayName: 'left pocket',
+    gridSpot: getGridSpotKey([4, 4]),
+    image: 'inventory-pocket.png',
+  },
+
+  r_store: {
+    displayName: 'right pocket',
+    gridSpot: getGridSpotKey([4, 5]),
+    image: 'inventory-pocket.png',
+  },
+};
+
+enum ObscuringLevel {
+  Completely = 1,
+  Hidden = 2,
+}
+
+type Interactable = {
+  interacting: BooleanLike;
+};
+
+/**
+ * Some possible options:
+ *
+ * null - No interactions, no item, but is an available slot
+ * { interacting: 1 } - No item, but we're interacting with it
+ * { icon: icon, name: name } - An item with no alternate actions
+ *   that we're not interacting with.
+ * { icon, name, interacting: 1 } - An item with no alternate actions
+ *   that we're interacting with.
+ */
+type StripMenuItem =
+  | null
+  | Interactable
+  | ((
+      | {
+          icon: string;
+          name: string;
+          alternate: string;
+        }
+      | {
+          obscured: ObscuringLevel;
+        }
+    ) &
+      Partial<Interactable>);
+
+type StripMenuData = {
+  items: Record<keyof typeof SLOTS, StripMenuItem>;
```

Comment:
`keyof typeof` is weird - this should just be the new type described above

Reply:
Done I think?